components/Results.js

```diff
@@ -46,26 +47,55 @@ export default class Results extends React.Component {
           </View>
         );
       } else {
-          console.log("There are ..." + this.state.dataSource.product.ingredients_from_palm_oil_tags.length())
+          // console.log(this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tags )
           //Problem here - the ingredients_from_palm_oil_n field doesn't exist on all records. It seems that ngredients_from_palm_oil_tags works though!
-          if(this.state.dataSource.product.ingredients_from_palm_oil_n == 0){
+          if (this.state.dataSource.product.ingredients_from_palm_oil_n > 0 || (this.state.dataSource.product.ingredients_from_palm_oil_tags != undefined && this.state.dataSource.product.ingredients_from_palm_oil_tags.length > 0)) {
           return(
           <View style={styles.container}>
             <Text style={styles.text}> {this.state.dataSource.product.product_name} </Text>
             <Text style={styles.text}> There are {this.state.dataSource.product.ingredients_from_palm_oil_n} ingredient from Palm oil in this </Text>
-            <Animatable.Text animation="zoomInUp"><Icon name="thumbs-up" size={150} color="#ffff"/></Animatable.Text>
+            <Animatable.Text animation="zoomInUp"><Icon name="thumbs-down" size={150} color="#ffff"/></Animatable.Text>
           </View>
           );
-          } else {
+          } else if (this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_n > 0 || 
+            
+            ( this.state.dataSource.product.ingredients_that_may_be_from_palm_oil_tabs != undefined 
```

Comment:
Typo here! tabs instead of tags which makes me think that these lines aren't used?